planet/js/GlobalCard.js

```diff
@@ -200,9 +200,30 @@ class GlobalCard {
         (!data.success && data.error === "ERROR_ACTION_NOT_PERMITTED") ?
             this.setLike(like) : this.setLike(like) ;
     };
+ setLike(like) {
+
+        function debounce(func, wait) {
+            let timeout;
+            return function () {
+                const context = this;
+                const args = arguments;
+                const later = function () {
+                    timeout = null;
```

Comment:
Hey @vivekbopaliya can you please tell why we have debounce function in the setLike method? 

Can we define debounce function outside of it? like below: 

`debounce(func, wait) {
    let timeout;
    return function (...args) {
        const context = this;
        const later = function () {
            timeout = null;
            func.apply(context, args);
        };
        clearTimeout(timeout);
        timeout = setTimeout(later, wait);
    };
}`

and in the setLike method: 

setLike(like) {
    if (!this.debouncedSetLike) {
        this.debouncedSetLike = this.debounce(() => {
            this.Planet.ProjectStorage.like(this.id, like);
        }, 500);
    }

    this.debouncedSetLike();
}`

This ensures that the debounce function is not redefined every time setLike is called, reducing redundancy.